rats-apps/docs/index.md

```diff
@@ -1,28 +1,202 @@
+    ```bash
+    poetry install
+    python -m foo
+    ```
+
+!!! tip
+    Any number of plugins can be registered this way. Extending the behavior of our
+    applications through plugins allows us to provide an opinionated, extensive default
+    experience, but allow advanced, unplanned uses to adjust most implementation details. The
+    `rats` project has a few built-in plugins for common application capabilities that we can
+    explore more later.
+
+Our plugin class is an implementation of the `apps.Container` protocol, which allows us to
+quickly make services available to other parts of the system. Let's finish moving our `main()`
+function into our plugin container and use `apps.SimpleApplication` to run it.
+
+=== ":material-language-python: ~/code/src/python/foo/_plugin.py"
+    ```python
+    from tempfile import mkdtemp
+    from rats import apps
+
+
+    @apps.autoscope
+    class PluginServices:
+        MAIN = ServiceId[apps.Executable]("main")
+        FOO_PATH = ServiceId[Path]("foo-path")
+
+
+    class PluginContainer(apps.Container):
+
+        _app: apps.Container
+
+        def __init__(self, app: apps.Container) -> None:
+            self._app = app
+
+        @apps.service(PluginServices.MAIN)
+        def _main(self) -> apps.Executable:
+            def main() -> None:
+                print("hello, world")
+                # we can access any service defined in a loaded plugin container.
+                print(f"foo path: {self._app.get(PluginServices.FOO_PATH)}")
+
+            return apps.App(main)
+
+        @apps.service(PluginServices.FOO_PATH)
+        def _foo_path(self) -> Path:
+            return Path(mkdtemp(dir=Path(".tmp")))
+    ```
+
+=== ":material-language-python: ~/code/src/python/foo/\_\_main\_\_.py"
+    ```python
+    from rats import apps
+    from ._plugin import PluginServices
+
+
+    def run() -> None:
+        app = apps.SimpleApplication("foo.plugins")
+        app.execute(PluginServices.MAIN)
+
+
+    if __name__ == "__main__":
+        run()
+    ```
+
+!!! success
+    We used `rats.app` to define a container with a couple services, and make a small
+    `__main__.py` file to allow us to run the service defined by `PluginServices.MAIN`.
+
+    You can create a script entry in your `pyproject.toml` file to expose your application through
+    a terminal command:
+
+    ```toml
+    [tool.poetry.scripts]
+    foo = "foo.__main__:run"
+    ```
+## Examples
+Most of the examples below can be executed by running the related module's `__main__.py` file.
+
+### rats.apps
 
 ```py linenums="1" title="src/python/rats/apps/__main__.py"
 --8<-- "rats-apps/src/python/rats/apps/__main__.py"
 ```
 
-## rats.annotations
+### rats.annotations
+This package helps create decorators meant to attach annotations to functions or methods. The
+annotation functions in `rats.apps` are a good example of using this package to create a friendly
+developer experience in your own domains.
 
```

Comment:
This is not well explained, I miss the point for annotating functions / classes / methods. You are run checking the annotations but not sure of the purpose. Is the goal to have a better UX in IDEs? Explain what the benefit is for attaching a namedtuple to a method...